### PR TITLE
Support for Scorpio in maint-1.0

### PR DESCRIPTION
**Pull request by jayeshkrishna**
Adding support for Scorpio (and Scorpio classic) in
maint-1.0

[BFB]

### CODE REVIEW

**Comment by jayeshkrishna** [comment]
All of the fixes included in this PR are already available on E3SM master. However some of the fixes had to be ported to be compatible with maint-1.0 .

This PR includes changes (some are minor) to,
* CIME
* CAM, CLM, CICE components
* New git submodules added for Scorpio

File: .gitmodules [hunk 1]
```diff
@@ -11,3 +11,11 @@
 [submodule "components/mpas-source"]
 	path = components/mpas-source
 	url = https://github.com/MPAS-Dev/MPAS-Model.git
+[submodule "externals/scorpio"]
+	path = externals/scorpio
+  url = git@github.com:E3SM-Project/scorpio.git
```
Comment (jgfouca) [inline .gitmodules]: Maint branches use https protocol for submodules.

Comment (jayeshkrishna) [reply .gitmodules]: ok, I will change that

Comment (jayeshkrishna) [reply .gitmodules]: This is fixed on the branch now

**Pull request merged**